src/gologin-api.js

```diff
@@ -0,0 +1,82 @@
+import puppeteer from 'puppeteer-core';
+import GoLogin from './gologin.js';
+
+export function getDefaultParams() {
+  return {
+    token: process.env.GL_API_TOKEN,
+    profile_id: process.env.GL_PROFILE_ID,
+    executablePath: process.env.GL_EXECUTABLE_PATH,
+  };
+}
+
+const createLegacyGologin = (params) => {
+  const defaults = getDefaultParams();
+  return new GoLogin({
+    ...defaults,
+    ...params,
+  });
+};
+
+export const delay = (ms = 250) => new Promise((res) => setTimeout(res, ms));
+
+export function GologinApi({ token }) {
+  if (!token) {
+    throw new Error('GoLogin API token is missing');
+  }
+  const browsers = [];
+  const legacyGls = [];
+
+  const launchLocal = async (params) => {
```

Comment:
this function only depends on `browsers` and `legacyGIs` so it also can be moved out of `GologinApi` class to avoid redundant resources use